x-pack/legacy/plugins/siem/public/components/navigation/index.tsx

```diff
@@ -22,14 +22,19 @@ import {
   networkModel,
 } from '../../store';
 import { CONSTANTS } from '../url_state/constants';
+import { RouteSpyState, useRouteSpy } from '../../utils/route/spy_routes';
 
-export class SiemNavigationComponent extends React.Component<
-  RouteComponentProps & TabNavigationProps
-> {
-  public shouldComponentUpdate(nextProps: Readonly<RouteComponentProps>): boolean {
+export class SiemNavigationComponent extends React.Component<TabNavigationProps & RouteSpyState> {
```

Comment:
I am not sure about that because we still need to be connected to the redux connect. I think that it can be converted to a hook when will update our redux to use hook